### PR TITLE
Rspec cart

### PR DESCRIPTION
**Pull request by truongdh1903**
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/???)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Comment by truongdh1903** [comment]
ready

File: spec/controllers/carts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  describe "#create" do
+    before do
+      @cart_params = {
+        "product_detail_id" => 1,
+        "quantity" => 1
+      }
+      post :create, params: {cart: @cart_params}
+    end
+    it "should create a cart item" do
```
Comment (ghost) [inline spec/controllers/carts_controller_spec.rb]: thêm trường hợp thất bại vào em

**Review by ghost — COMMENTED** [review]
cmt

**Comment by truongdh1903** [comment]
ready

File: spec/controllers/carts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
```
Comment (ghost) [inline spec/controllers/carts_controller_spec.rb]: context của anh đâu

**Review by ghost — COMMENTED** [review]
cmt

File: spec/controllers/carts_controller_spec.rb (hunk 2 above)
Comment (truongdh1903) [reply spec/controllers/carts_controller_spec.rb]: T_T

**Comment by truongdh1903** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**